src/librbd/AioObjectRequest.cc

```diff
@@ -434,12 +434,12 @@ namespace librbd {
   }
 
   bool AbstractAioObjectWrite::send_post() {
-    RWLock::RLocker owner_locker(m_ictx->owner_lock);
-    RWLock::RLocker snap_locker(m_ictx->snap_lock);
```

Comment:
I believe ```snap_lock``` is used to protect ```m_ictx->object_map``` during image refreshes, so it can't just be moved.      You might need to do something like this to avoid the lock recursion: https://github.com/ceph/ceph/blob/master/src/librbd/AioObjectRequest.cc#L219